lib/recipes-data/src/lib/extract-recipes.ts

```diff
@@ -3,31 +3,35 @@ import type {Blocks} from "@guardian/content-api-models/v1/blocks";
 import type {Content} from "@guardian/content-api-models/v1/content";
 import {ContentType} from "@guardian/content-api-models/v1/contentType";
 import {ElementType} from "@guardian/content-api-models/v1/elementType";
+import {registerMetric} from "@recipes-api/cwmetrics";
 import type {RecipeReferenceWithoutChecksum} from './models';
 
-export function extractAllRecipesFromArticle(content: Content): RecipeReferenceWithoutChecksum[] {
+export async function extractAllRecipesFromArticle(content: Content): Promise<RecipeReferenceWithoutChecksum[]> {
   if (content.type == ContentType.ARTICLE && content.blocks) {
     const articleBlocks: Blocks = content.blocks
-    const getAllMainBlockRecipesIfPresent: RecipeReferenceWithoutChecksum[] = extractRecipeData(content.id, articleBlocks.main as Block)
+    const getAllMainBlockRecipesIfPresent = extractRecipeData(content.id, articleBlocks.main as Block)
     const bodyBlocks = articleBlocks.body as Block[]
-    const getAllBodyBlocksRecipesIfPresent: RecipeReferenceWithoutChecksum[] = bodyBlocks.flatMap(bodyBlock => extractRecipeData(content.id, bodyBlock))
-    return getAllMainBlockRecipesIfPresent.concat(getAllBodyBlocksRecipesIfPresent)
+    const getAllBodyBlocksRecipesIfPresent = bodyBlocks.flatMap(bodyBlock => extractRecipeData(content.id, bodyBlock))
+    const recipes = getAllMainBlockRecipesIfPresent.concat(getAllBodyBlocksRecipesIfPresent)
+    const failureCount = recipes.filter(recp => !recp).length
+    await registerMetric("FailedRecipes", failureCount)
+    await registerMetric("SuccessfulRecipes", recipes.length)
```

Comment:
👍

Reply:
You should test that `registerMetric()` is properly called in the tests for extract-recipes.ts